crypto/rand/prov_seed.c

```diff
@@ -130,3 +136,9 @@ void ossl_rand_cleanup_nonce(ossl_unused OSSL_LIB_CTX *ctx,
 {
     OPENSSL_clear_free(buf, len);
 }
+
+void ossl_rand_cleanup_user_nonce(ossl_unused OSSL_LIB_CTX *ctx,
+                                  unsigned char *buf, size_t len)
+{
+    OPENSSL_clear_free(buf, len);
+}
```

Comment:
Without the changes from #22394, the implementations  of  `ossl_rand_cleanup_user_nonce` and `ossl_rand_cleanup_nonce`  are identical. It's important however, that both callbacks exist.

Now  #22394 will depend on this pull request to be merged first.